### DISCUSSION TITLE
Full Cone nat support?

### DISCUSSION THREAD

**Discussion by redpoisonv3** [fabianishere/udm-kernel-tools#61] [General]
i wanted to ask if this kernel adds support for full cone nat, since the stock firmware currently doesn't have it enabled.

**Comment by fabianishere** [comment]
Which functionality in the kernel are you referring to (that’s needed to enable full cone NAT)?

This page suggests it should be doable with iptables only: http://www.joewein.net/info/sw-iptables-full-cone-nat.htm

**Reply by redpoisonv3** [reply]
i was under the impression a specific kernel version was required, i wasn't sure if the sock kernel had support for it, thank you.

**Reply by redpoisonv3** [reply]
question how would i keep it persitant even if the udm is rebooted?

**Reply by redpoisonv3** [reply]
just a little stuck on how to adapt the rules to the udm

**Reply by redpoisonv3** [reply]
i wanted to be sure that the udm isnt missing something like this in the stock kernel
https://github.com/Chion82/netfilter-full-cone-nat

**Reply by fabianishere** [reply]
Sorry, I am unable to advice you on this since I don't have any experience with setting up a full cone NAT. I reckon you don't need a custom kernel module for this, but if you want, you can build that kernel module yourself with the kernel sources at https://github.com/fabianishere/udm-kernel.

> question how would i keep it persitant even if the udm is rebooted?

You can create a custom boot script to create the IPTables rules on boot: https://github.com/boostchicken-dev/udm-utilities/tree/master/on-boot-script

**Reply by redpoisonv3** [reply]
all good thank you for your help i really appreciate it
